blocks/form/checkbox.js

```diff
@@ -0,0 +1,27 @@
+function update(checkboxgroup, isChecked) {
+  if (isChecked) { // remove required if any of them is checked
+    checkboxgroup.forEach((checkbox) => checkbox.removeAttribute('required'));
+  } else if (checkboxgroup.every((checkbox) => !checkbox.checked)) {
+    // set required if none of them is checked
+    checkboxgroup.forEach((checkbox) => checkbox.setAttribute('required', ''));
+  }
+}
+
+export default async function decorate(form) {
```

Comment:
you could probably remove the `event.target.checked` and simplify the `update` signature, effectively making it a toggle-behavior. 

also, `Array.from` makes the code more readable:

```
export default async function decorate(form) {
  const checkboxes = Array.from(form.querySelectorAll('input[type=checkbox][required]'));
  const checkboxNameMap = checkboxes.reduce((map, checkbox) => {
    if (map.has(checkbox.name)) {
      map.get(checkbox.name).push(checkbox);
    } else {
      map.set(checkbox.name, [checkbox]);
    }
    return map;
  }, new Map());

  Array.from(checkboxNameMap.values())
    .filter((checkboxgroup) => checkboxgroup.length > 1)
    .forEach((checkboxgroup) => {
      checkboxgroup.forEach((checkbox) => {
        checkbox.onchange = () => update(checkboxgroup);
      });
    });
}
```